TFNetworkLayer.py

```diff
@@ -437,6 +437,8 @@ def transform_config_dict(cls, d, network, get_layer):
           # Not resolving this in the dict as target, but call get_layer to make it available.
           if target.startswith("layer:"):
             target_layers[target] = get_layer(target[len("layer:"):])
+          else:
+            network.used_data_keys.add(target)
```

Comment:
Maybe add a comment, why we do this.

Reply:
The comment also should explain why this is a workaround, and what probably a better solution would be.

Reply:
can you provide a text? because I do not fully understand neither the problem nor the possible fix

Reply:
Make sure that these do not exceed 120 chars.

> Note: This is a workaround for cases where we need to know about used data keys before the layer itself is constructed
> (e.g. in `_SubnetworkRecCell.get_output`).
> A nicer solution would be to not modify this here,
> but instead lazily handle it in `TFNetwork.get_extern_data`,
> such that we do not need to know in advance which data keys we need.